### ISSUE TITLE
`SING_SINGLETON_GETTER_NOT_SYNCHRONIZED` not reported when the lazy init uses a ternary instead of an if-statement

### ISSUE THREAD

**Issue by Chordrain** [spotbugs/spotbugs#4154]
### SpotBugs version

SpotBugs 4.9.8 (JDK 21, javac 21, source/target 21)

### Check name

`SING_SINGLETON_GETTER_NOT_SYNCHRONIZED`

### Description

The detector flags an unsynchronized singleton getter that lazily initializes its static instance field with an `if (instance == null) { instance = new ...(); }` block, but it stops reporting when the same lazy initialization is written as the equivalent ternary assignment `instance = (instance == null) ? new ...() : instance;`. Both forms are unsynchronized lazy initializations of the same static `instance` field inside the same static getter — exactly the anti-pattern the rule targets — and both have the same thread-safety hazard. The detector appears to pattern-match the `if`-statement shape of the lazy init and misses the ternary shape.

### Reproducer

```java
// BEFORE — if-statement lazy init, 1 report
public class Issue3280 {
    private static volatile Issue3280 instance = null;

    public static Issue3280 getInstance() {
        if (null == instance) {
            instance = new Issue3280();
        }
        instance.factorial(5);
        return instance;
    }

    private Issue3280() {
    }

    private int factorial(int n) {
        if (n == 0) {
            return 1;
        }
        return n * factorial(n - 1);
    }
}
```

```java
// AFTER — equivalent ternary lazy init, 0 reports
public class Issue3280 {
    private static volatile Issue3280 instance = null;

    public static Issue3280 getInstance() {
        instance = null == instance ? new Issue3280() : instance;
        instance.factorial(5);
        return instance;
    }

    private Issue3280() {
    }

    private int factorial(int n) {
        if (n == 0) {
            return 1;
        }
        return n * factorial(n - 1);
    }
}
```

### Expected behavior

`SING_SINGLETON_GETTER_NOT_SYNCHRONIZED` should fire on both forms, since both are unsynchronized lazy initializations of the singleton instance field in an unsynchronized getter.

### Actual behavior

- BEFORE (if-statement): 1 report — `Instance-getter method of class using singleton design pattern (Issue3280) is not synchronized.`
- AFTER (ternary): 0 reports.
